### ISSUE TITLE
extended networking flake: subnet: Services should serve a basic endpoint from pods

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#9355]
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_networking/1301/ failed subnet tests

```
 Extended.[k8s.io] Services should serve a basic endpoint from pods [Conformance]   1 min 8 sec 1
 Extended.[k8s.io] Services should serve multiport endpoints from pods [Conformance]    1 min 27 sec    1
 Extended.[k8s.io] Services should be able to create a functioning NodePort service 19 sec  1
 Extended.[k8s.io] EmptyDir volumes should support (root,0644,tmpfs) [Conformance]  5 min 7 sec 1
 Extended.[k8s.io] Networking should function for intra-pod communication [Conformance] 5 min 27 sec    1
 Extended.[k8s.io] Networking [k8s.io] [Flaky] Granular Checks should function for pod communication between nodes  5 min 25 sec    1
 Extended.[networking] services basic functionality should allow connections to another pod on a different node via a service IP    5 min 24 sec    1
 Extended.[networking] services when using a single-tenant plugin should allow connections to pods in different namespaces on different nodes via service IPs   5 min 32 sec    1
 Extended.[networking] network isolation when using a single-tenant plugin should allow communication between pods in different namespaces on different nodes
```

Due to what looks like a race / failure to setup the subnet plugin.

```
Jun 15 11:02:25.449: INFO: Pod e2e-tests-services-ctzd1 pod1    nettest-node-2  <nil>
Jun 15 11:02:25.449: FAIL: Timed out waiting for service endpoint-test2 in namespace e2e-tests-services-ctzd1 to expose endpoints map[pod1:[80]] (1m0s elapsed)
[1mSTEP[0m: Collecting events from namespace "e2e-tests-services-ctzd1".
Jun 15 11:02:25.509: INFO: At 2016-06-15 11:01:25 -0400 EDT - event for pod1: {default-scheduler } Scheduled: Successfully assigned pod1 to nettest-node-2
Jun 15 11:02:25.509: INFO: At 2016-06-15 11:01:25 -0400 EDT - event for pod1: {kubelet nettest-node-2} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "POD" with RunContainerError: "runContainer: API error (404): failed to create endpoint k8s_POD.d1f7ffed_pod1_e2e-tests-services-ctzd1_079d3b8d-330a-11e6-97db-0242a1568a38_a1b252d2 on network bridge: adding interface veth4a36ab6 to bridge docker0 failed: could not find bridge docker0: route ip+net: no such network interface\n"

Jun 15 11:02:25.509: INFO: At 2016-06-15 11:01:41 -0400 EDT - event for pod1: {kubelet nettest-node-2} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "POD" with RunContainerError: "runContainer: API error (404): failed to create endpoint k8s_POD.d1f7ffed_pod1_e2e-tests-services-ctzd1_079d3b8d-330a-11e6-97db-0242a1568a38_4860bbdb on network bridge: adding interface veth7302cdc to bridge docker0 failed: could not find bridge docker0: route ip+net: no such network interface\n"

Jun 15 11:02:25.509: INFO: At 2016-06-15 11:01:54 -0400 EDT - event for pod1: {kubelet nettest-node-2} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "POD" with RunContainerError: "runContainer: API error (404): failed to create endpoint k8s_POD.d1f7ffed_pod1_e2e-tests-services-ctzd1_079d3b8d-330a-11e6-97db-0242a1568a38_ac941c6f on network bridge: adding interface veth1f5123a to bridge docker0 failed: could not find bridge docker0: route ip+net: no such network interface\n"

Jun 15 11:02:25.509: INFO: At 2016-06-15 11:02:09 -0400 EDT - event for pod1: {kubelet nettest-node-2} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "POD" with RunContainerError: "runContainer: API error (404): failed to create endpoint k8s_POD.d1f7ffed_pod1_e2e-tests-services-ctzd1_079d3b8d-330a-11e6-97db-0242a1568a38_846bba44 on network bridge: adding interface veth6998bda to bridge docker0 failed: could not find bridge docker0: route ip+net: no such network interface\n"

Jun 15 11:02:25.509: INFO: At 2016-06-15 11:02:23 -0400 EDT - event for pod1: {kubelet nettest-node-2} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "POD" with RunContainerError: "runContainer: API error (404): failed to create endpoint k8s_POD.d1f7ffed_pod1_e2e-tests-services-ctzd1_079d3b8d-330a-11e6-97db-0242a1568a38_1b9ba95b on network bridge: adding interface veth50f4195 to bridge docker0 failed: could not find bridge docker0: route ip+net: no such network interface\n"

Jun 15 11:02:25.512: INFO: POD   NODE            PHASE    GRACE  CONDITIONS
Jun 15 11:02:25.512: INFO: pod1  nettest-node-2  Pending  30s    [{Ready False 0001-01-01 00:00:00 +0000 UTC 2016-06-15 11:01:25 -0400 EDT ContainersNotReady containers with unready status: [test]}]
```

**Comment by smarterclayton** [comment]
In https://github.com/openshift/origin/pull/7918

**Comment by danwinship** [comment]
So the logs don't show how we got into this situation (which appears to be that docker never got restarted and told to use lbr0). At least somewhat related to https://trello.com/c/LPDKU4Nz though (which is that we don't wait for docker to finish restarting after we restart it).

**Comment by smarterclayton** [comment]
Ok, thanks for investigating.  We can bump to p2

**Comment by danwinship** [comment]
@dcbw: once we switch to being CNI-based we won't need to edit $DOCKER_NETWORK_ARGS and restart docker any more, right?

**Comment by danwinship** [comment]
> @dcbw: once we switch to being CNI-based we won't need to edit $DOCKER_NETWORK_ARGS and restart docker any more, right?

Replying to myself, yes, so this should be fixed by https://github.com/openshift/openshift-sdn/pull/320

**Comment by dcbw** [comment]
@danwinship yes https://github.com/openshift/openshift-sdn/pull/320 will fix the issue.

@smarterclayton in this case, openshift-node has overflowed the systemd journal limits.  Pretty hard to figure out what the issue is unless we could (a) get more logs (which might not even help), or (b) poke around the system...

**Comment by danwinship** [comment]
This was fixed by #11082 though there's now a different failure in "Services should serve a basic endpoint from pods"
